src/workspace.ts

```diff
@@ -208,9 +227,52 @@ export class Workspace implements Disposable, QuickDiffProvider {
       sourceControlResources.push(new PlasticScmResource(changeInfo, this));
     }
 
+    if (this.mConfig.consolidateUnrealOneFilePerActorChanges) {
+      const unrealLevelNames: string[] = [];
+      for (const changeInfo of changeInfos) {
+        let skipFile =
+          unrealOfpaRegex.exec(changeInfo.path.fsPath) === null ||
+          !changeInfo.path.fsPath.endsWith("uasset");
+
+        for (const name of unrealLevelNames) {
+          if (changeInfo.path.fsPath.includes(name)) {
+            skipFile = true;
+            break;
+          }
+        }
+
+        if (skipFile) {
+          continue;
+        }
+
+        const pathParts = changeInfo.path.fsPath.replace(/\\/g, "/").split("/");
+        const ofpaIndex = pathParts.findIndex((value: string) => unrealOfpaRegex.exec(value) !== null);
+        const levelRelativeLocation = pathParts.slice(ofpaIndex + 1, pathParts.length - 3).join("/");
+        const pathToContentDir = pathParts.slice(0, ofpaIndex).join("/");
+        const levelName = path.basename(levelRelativeLocation);
+
+        unrealLevelNames.push(levelName);
+
+        const resourceState: SourceControlResourceState = {
+          decorations: {
+            faded: false,
+            strikeThrough: false,
+            tooltip: "Hello",
+          },
+          resourceUri: Uri.from({
+            path: `${pathToContentDir}/${levelRelativeLocation}.umap`,
+            scheme: "file",
+          }),
+        };
+
+        unrealLevelResources.push(resourceState);
+      }
+    }
+
     this.mStatusResourceGroup.resourceStates = sourceControlResources;
-    this.mSourceControl.count = changeInfos.filter(
-      changeInfo => changeInfo.type !== ChangeType.Private).length;
+    this.mUnrealLevelsResourceGroup.resourceStates = unrealLevelResources;
+    this.mSourceControl.count = sourceControlResources.length + unrealLevelResources.length;
```

Comment:
This is changing the behavior of this "number of files changed" indicator:

- until now was telling the user how many files would be checked-in
- now it will also list files that are private, that won' be checked-in

Reply:
Ultimately, we like the end result :)